packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -126,6 +141,62 @@ export class QueuedRequestController extends BaseController<
     return this.state.queuedRequestCount;
   }
 
+  /**
+   * Switch the current globally selected network if necessary for processing the given
+   * request.
+   *
+   * @param request - The request currently being processed.
+   * @throws Throws an error if the current selected `networkClientId` or the
+   * `networkClientId` on the request are invalid.
+   */
+  async #switchNetworkIfNecessary(
+    request: QueuedRequestMiddlewareJsonRpcRequest,
+  ) {
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    if (request.networkClientId === selectedNetworkClientId) {
+      return;
+    }
+
+    const toNetworkConfiguration = this.messagingSystem.call(
+      'NetworkController:getNetworkConfigurationByNetworkClientId',
+      request.networkClientId,
+    );
+    const fromNetworkConfiguration = this.messagingSystem.call(
+      'NetworkController:getNetworkConfigurationByNetworkClientId',
+      selectedNetworkClientId,
+    );
+    if (!toNetworkConfiguration) {
+      throw new Error(
+        `Missing network configuration for ${request.networkClientId}`,
+      );
+    } else if (!fromNetworkConfiguration) {
+      throw new Error(
+        `Missing network configuration for ${selectedNetworkClientId}`,
+      );
+    }
+
+    const requestData = {
+      toNetworkConfiguration,
+      fromNetworkConfiguration,
+    };
+    await this.messagingSystem.call(
+      'ApprovalController:addRequest',
+      {
+        origin: request.origin,
+        type: ApprovalType.SwitchEthereumChain,
+        requestData,
+      },
+      true,
```

Comment:
This would result in `enqueueRequest` throwing an error. This is tested by the `when the network switch for a single request fails` describe block in the controller tests. Then this error would be caught by the middleware and returned to the dapp making the request.

Reply:
Ok gotcha I missed that new test